tests/test_metadata/test_v3.py

```diff
@@ -304,6 +304,19 @@ def test_metadata_to_dict(
     assert observed == expected
 
 
+def test_json_indent():
+    m = GroupMetadata()
+    d = m.to_buffer_dict(default_buffer_prototype())["zarr.json"].to_bytes()
+    class TestIndentEncoder(json.JSONEncoder):
+        def __init__(self, *args: Any, **kwargs: Any) -> None:
+            super().__init__(*args, **kwargs)
+            self.indent = 2
+
+    # expected has extra ' ' on each line compared with json.dumps( indent=2)
+    expected = json.dumps(json.loads(d), cls=TestIndentEncoder).encode()
```

Comment:
`json.dumps` takes an `indent` keyword argument, so I don't think we need the custom encoder class for this test

Reply:
Yeah, I tried that first!
But `json.dumps(data, indent=2)` gives a different output than a class derived from `json.JSONEncoder`. The `json.JSONEncoder` has an extra whitespace at the end of each line (see my comment on line 315 above).
I don't know why this is (I guess it would be nice to know)?!

Reply:
weird, i would also like to know why this is the case. is the extra whitespace more or less correct?

Reply:
Just confirmed this doing a test writing of some zarr data. All the `zarr.json` have whitespace at the end of rows that end with a comma.

Reply:
> weird, i would also like to know why this is the case. is the extra whitespace more or less correct?

I would say it's not really correct to have the extra whitespace at the end of lines.
Here, there shouldn't be whitespace after `"foo",`

```
>>> json.dumps({"test": "foo", "bar": 2}, cls=TestIndentEncoder)
'{\n  "test": "foo", \n  "bar": 2\n}'
```